src/controllers/__tests__/request.test.js

```diff
@@ -0,0 +1,278 @@
+
+      expect(response.header['content-type']).toContain('application/json');
+      expect(response.statusCode).toBe(200);
+      expect(responseBody).toMatchObject(trueRequest);
+      expect(responseBody._id.toString()).toBe(RequestId.toString());
+    });
+
+    test('Should respond with an error message when the request id is not a valid ObjectId', async () => {
+      const response = await request(server).get(
+        `/api/requests/${invalidRequestId}`
+      );
+      const responseBody = response.body;
+
+      expect(response.statusCode).toBe(422);
+      expect(responseBody.message).toBeDefined();
+      expect(responseBody.message).toBe('Requested request ID is not valid!');
+    });
+
+    test('Should return an error message when the request does not exist', async () => {
+      const response = await request(server).get(
+        `/api/requests/${nonExistingRequestId}`
+      );
+      const responseBody = response.body;
+
+      expect(response.statusCode).toBe(422);
+      expect(responseBody.message).toBeDefined();
+      expect(responseBody.message).toBe(
+        'The request with the specified ID was not found.'
+      );
+    });
+  });
+
+  describe('PUT /api/requests/:id', () => {
+    test('Should update name on matching request', async () => {
+      const response = await request(server)
+        .put(`/api/requests/${RequestId}`)
+        .set('Cookie', authCookie)
+        .send({ name: 'Updated name' });
+      const responseBody = response.body;
+
+      expect(response.header['content-type']).toContain('application/json');
+      expect(response.statusCode).toBe(200);
+      expect(responseBody.name).toBe('Updated name');
+    });
+
+    test('Should update description on matching request', async () => {
+      const response = await request(server)
+        .put(`/api/requests/${RequestId}`)
+        .set('Cookie', authCookie)
+        .send({ description: 'Updated description' });
+      const responseBody = response.body;
+
+      expect(response.header['content-type']).toContain('application/json');
+      expect(response.statusCode).toBe(200);
+      expect(responseBody.description).toBe('Updated description');
+    });
+
+    test('Should respond with an error message when to be updated request ID not found', async () => {
+      const response = await request(server)
+        .put(`/api/requests/${nonExistingRequestId}`)
+        .set('Cookie', authCookie);
+      const responseBody = response.body;
+
+      expect(response.statusCode).toBe(422);
+      expect(responseBody.message).toBeDefined();
+      expect(responseBody.message).toBe(
+        'The request with the specified ID was not found.'
+      );
+    });
+  });
+
+  describe('DELETE /api/requests/:id', () => {
+    test('Should delete a request', async () => {
+      const response = await request(server)
+        .delete(`/api/requests/${RequestId}`)
+        .set('Cookie', authCookie);
+
+      expect(response.statusCode).toBe(204);
+    });
```

Comment:
I forgot to mention one important thing in the review. Could you add another test under this test to make sure that the request with the specified id was deleted? here you are only checking for the response code, you can send a get by id request and expect an error